Library/Homebrew/extend/os/linux/extend/ENV/super.rb

```diff
@@ -7,6 +7,7 @@ def self.bin
   # @private
   def setup_build_environment(formula = nil)
     generic_setup_build_environment(formula)
+    self["HOMEBREW_OPTIMIZATION_LEVEL"] = "O2"
```

Comment:
this might get overwritten by the generic `setup_build_environment` since it calls `super` before setting this variable to `"Os"`:

https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/ENV/super.rb#L39-L48

Reply:
Our `setup_build_environment`, which calls`generic_setup_build_environment`, where the `super` calls `setup_build_environment` in `extend/ENV/shared.rb`, so it behaves as desired.
https://github.com/Homebrew/brew/blob/0af24cf8bc6ae9572de19c1f1209320f81cf31e2/Library/Homebrew/extend/ENV/shared.rb#L30